### PR TITLE
fix: skip socket registry in contract bindings

### PR DESCRIPTION
**Pull request by hopeyen**
## Why are these changes needed?

Don't error out when the socket registry is not deployed

! This should be reverted once the socket registry contract is deployed

## Checks

- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- [ ] I've checked the new test coverage and the coverage percentage didn't drop.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: core/eth/reader.go
Comment (jianoaix) [inline core/eth/reader.go:174]: Can you use a flag to control these?

Comment (hopeyen) [reply core/eth/reader.go:174]: I could! Although it looks like I would need to update NewReader to take in a flag, and that requires updating flags for a lot of components like the controller, retriever, dataapi, apiserver, disperser, ... Do you think it is still the best way to process?

Comment (jianoaix) [reply core/eth/reader.go:174]: It looks the easier way is to hold off the socket registry change until the contract is live. Is there anything blocked by not merging the change?

Comment (hopeyen) [reply core/eth/reader.go:174]: I agree that's the easier way. To my understanding this PR unblocks this week's release but the contract is going live at the end of the sprint. 
Alternatively, we revert the changes that adds the socket registry and go ahead with the release; but then we wanted to do decentralized retrieval this sprint, and devops is using semvar to check correctness of the socket registry (migration from thegraph indexer)

Comment (jianoaix) [reply core/eth/reader.go:174]: I was talking about not merging previous PR which added in this part of code.
It sounds nothing is blocked by this PR; all what you mentioned I'd say the blocker is the the deployment of the contract.

Comment (jianoaix) [reply core/eth/reader.go:174]: A more appropriate way seems still to be some CLI flags, maybe defined similar to the graph indexer with default to false, passed in as a config etc. It'd be a standard approach as we add other contracts in the future.

To unblock the release for now, what if you add a local flag (not a CLI flag), to control flow here and the below GetOperatorSocket? Then when you actually ready to enable it, there is only one place to change, instead of editing code around.

Comment (hopeyen) [reply core/eth/reader.go]: yep, that sounds great. I added a local variable `socketRegistryEnabled` and used that for control flow here and `GetOperatorSocket`

**Review by jianoaix — APPROVED** [review]

**Review by pschork — APPROVED** [review]

**Pull request merged**
